modules/claims_api/app/sidekiq/claims_api/flash_updater.rb

```diff
@@ -33,7 +33,30 @@ def persist_exception(e, auto_claim_id: nil, message: { key: e.code, text: e.mes
       log_exception_to_sentry(e)
     end
 
+    def add_flash(user, flash_name)
+      if Flipper.enabled? :claims_api_flash_updater_uses_local_bgs
+        bgs_service(user).add_flash(file_number: user['ssn'], flash: { flash_name: })
+      else
+        bgs_service(user).add_flash(file_number: user['ssn'], flash_name:)
+      end
+    end
+
     def bgs_service(user)
+      if Flipper.enabled? :claims_api_flash_updater_uses_local_bgs
+        claimant_service(user)
+      else
+        bgs_ext_service(user).claimant
+      end
+    end
```

Comment:
Could we consolidate this logic? e.g.,

```rb
def add_flash(user, flash_name)
  if Flipper.enabled?(:claims_api_flash_updater_uses_local_bgs)
    service = claimant_service(user)
    service.add_flash(file_number: user['ssn'], flash: { flash_name: })
  else
    service = bgs_ext_service(user).claimant
    service.add_flash(file_number: user['ssn'], flash_name:)
  end
end 

```

Reply:
Oh never mind, I see we use `bgs_service` on line 16. We could cache the FF response so we donʼt check the db twice, but not super important:

```rb
def use_local_bgs?
   @use_local_bgs ||= Flipper.enabled?(:claims_api_flash_updater_uses_local_bgs)
end
```